contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -103,6 +106,23 @@ func (k *k8smScheduler) killTask(taskId string) error {
 	return err
 }
 
+func (k *k8smScheduler) terminateTask(task *podtask.T) error {
+	terminateMessage := struct {
+		ID                         string
+		DeletionTimestamp          *util.Time `json:"deletionTimestamp,omitempty"`
+		DeletionGracePeriodSeconds *int64     `json:"deletionGracePeriodSeconds,omitempty"`
+	}{task.ID, task.Pod.DeletionTimestamp, task.Pod.DeletionGracePeriodSeconds}
+
+	msg, err := json.Marshal(terminateMessage)
+	if err != nil {
+		return err
+	}
+
+	data := fmt.Sprintf("%s:%s", messages.TaskTerminate, string(msg)) //TODO(jdef) use a real message type
+	_, err = k.internal.driver.SendFrameworkMessage(task.Executor.GetExecutorId(), mutil.NewSlaveID(task.Spec.SlaveID), data)
```

Comment:
An alternative would be to watch the pods for a node in the executor and notify the kubelet about them. This is essentially what the kubelet would do if we used an apiserver config source. I think I would prefer that over this solution in the scheduler.